server/sonar-server/src/test/java/org/sonar/server/rule/RegisterRulesTest.java

```diff
@@ -842,6 +842,50 @@ public void declaring_a_rule_with_a_deprecated_RuleKey_still_used_should_throw_I
     });
   }
 
+  @Test
+  public void updating_the_parent_of_a_deprecated_rule_key_should_work() {
```

Comment:
it would help to identify what's changing from one execute to another if, for example, you used some methods to hide the repetitive/noise code from the one that matters and some variables/constants